### PR TITLE
infracost: init at 0.7.20

### PR DESCRIPTION
**Pull request by 06kellyjac**
###### Motivation for this change

WIP PR to get some thoughts.
Not 100% sure that the completion is working
Might want to modify this to have a `generic.nix` rather than a `v1.nix` and `v2.nix`

Package infracost, init at `0.7.20`
Add completions for go binaries using `urfave/cli`

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 114402` at 741e41ce run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/5886f7be89fb557fb6be83f6d54a0ec6">infracost</a></li></li>
    <li><a href="https://gist.github.com/77e4b86acfc040d3c65c0ccd2fc21cba">urfave-cli-completion</a></li></li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook postInstall`.

Near pkgs/development/libraries/urfave-cli-completion/default.nix:13:3:
```
   |
13 |   installPhase = let
   |   ^
```
</li>
    <li>warning: unused-argument

Unused argument: `fetchFromGitHub`.
Near pkgs/development/libraries/urfave-cli-completion/default.nix:1:16:
```
  |
1 | { lib, stdenv, fetchFromGitHub, writeShellScript, urfave-cli-completion_1, urfave-cli-completion_2 }:
  |                ^
```
</li>
  </ul>
</details>

File: pkgs/development/libraries/urfave-cli-completion/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  installPhase = let
+    linkCompletionsScript = writeShellScript "linkCompletions" ''
+      VER="$1"; NAME="$2"; OUT="$3"
+      DIR="$( cd "$( dirname "''${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+      mkdir -p $OUT/share/bash-completion/completions $OUT/share/zsh/site-functions
+      # Our typical $NAME.bash format wouldn't work with the script
+      ln -s "$DIR/$VER/bash_autocomplete" \
+            "$out/share/bash-completion/completions/$NAME"
+      ln -s "$DIR/$VER/zsh_autocomplete" \
+            "$out/share/zsh/site-functions/_$NAME"
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/urfave-cli-completion/default.nix:22]:
```suggestion
      ln -s "$DIR/$VER/bash_autocomplete" "$out/share/bash-completion/completions/$NAME"
      ln -s "$DIR/$VER/zsh_autocomplete" "$out/share/zsh/site-functions/_$NAME"
```

File: pkgs/development/libraries/urfave-cli-completion/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    '';
+  in
+  ''
+    runHook preInstallCheck
+    mkdir -p $out/share/urfave-cli-complete
+    ln -s ${urfave-cli-completion_2}/share/urfave-cli-complete/v2 $out/share/urfave-cli-complete
+    ln -s ${urfave-cli-completion_1}/share/urfave-cli-complete/v1 $out/share/urfave-cli-complete
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/urfave-cli-completion/default.nix:29]: Isn't this overwriting the link?

File: pkgs/development/libraries/urfave-cli-completion/v1.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  name = "${pname}-${version}";
+  owner = "urfave";
+  repo = "cli";
+  rev = "v${version}";
+  sha256 = "sha256-ZR3f7aFl0F9pZp/Bz2jpxhVrpkn2VnF6alg2Zc7B46o=";
+
+  postFetch = ''
+    tar xf $downloadedFile --strip=1
+    # remove shebang
+    sed -i '1,2d' autocomplete/bash_autocomplete
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/urfave-cli-completion/v1.nix:16]: Can we use substituteInPlace here?

File: pkgs/development/libraries/urfave-cli-completion/v1.nix [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+  meta = with lib; {
+    homepage = "https://github.com/urfave/cli/";
+    description = "Auto-completion files for urfave/cli powered tools";
+    license = licenses.mit;
+    maintainers = with maintainers; [ jk ];
+    platforms = platforms.all;
+  };
+}
+
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/urfave-cli-completion/v1.nix:28]: Extra final new line.

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -8636,6 +8638,12 @@ in
 
   urdfdom-headers = callPackage ../development/libraries/urdfdom-headers {};
 
+  urfave-cli-completion_2 = callPackage ../development/libraries/urfave-cli-completion/v2.nix {};
+  urfave-cli-completion_1 = callPackage ../development/libraries/urfave-cli-completion/v1.nix {};
+
+  # urfave-cli-completion = urfave-cli-completion_2;
+  urfave-cli-completion = callPackage ../development/libraries/urfave-cli-completion {};
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:8645]:
```suggestion
  urfave-cli-completion_2 = callPackage ../development/libraries/urfave-cli-completion/v2.nix { };
  urfave-cli-completion_1 = callPackage ../development/libraries/urfave-cli-completion/v1.nix { };

  # urfave-cli-completion = urfave-cli-completion_2;
  urfave-cli-completion = callPackage ../development/libraries/urfave-cli-completion { };
```

**Pull request closed without merging**